p2p/server.go

```diff
@@ -71,39 +60,60 @@ func (*peer) Name() string {
 func (s *peer) Start(ctx context.Context, wg *sync.WaitGroup, startupErr chan<- error) {
 	defer wg.Done()
 
-	if s.config.GetP2PPort() == 0 {
+	nc, err := s.config.GetConfig()
+	if err != nil {
+		startupErr <- err
+		return
+	}
+
+	if nc.GetP2PPort() == 0 {
 		startupErr <- errors.New("please provide a port to bind on")
 		return
 	}
 
 	// Make a host that listens on the given multiaddress
 	// first obtain the keys configured
-	priv, pub, err := s.createSigningKey()
+	priv, pub, err := s.createSigningKey(nc.GetSigningKeyPair())
 	if err != nil {
 		startupErr <- err
 		return
 	}
-	s.host, err = makeBasicHost(priv, pub, s.config.GetP2PExternalIP(), s.config.GetP2PPort())
+	s.host, err = makeBasicHost(priv, pub, nc.GetP2PExternalIP(), nc.GetP2PPort())
 	if err != nil {
 		startupErr <- err
 		return
 	}
 
-	s.mes = newP2PMessenger(ctx, s.host, s.config.GetP2PConnectionTimeout())
+	s.mes = newP2PMessenger(ctx, s.host, nc.GetP2PConnectionTimeout())
 	handler := s.handlerCreator()
 	s.mes.addHandler(pb.MessageType_MESSAGE_TYPE_SEND_ANCHORED_DOC, handler.HandleSendAnchoredDocument)
 	s.mes.addHandler(pb.MessageType_MESSAGE_TYPE_REQUEST_SIGNATURE, handler.HandleRequestDocumentSignature)
-	s.mes.init(CentrifugeProtocol)
+
+	tcs, err := s.config.GetAllTenants()
+	if err != nil {
+		startupErr <- err
+		return
+	}
+	var protocols []protocol.ID
+	for _, t := range tcs {
+		CID, err := identity.ToCentID(t.IdentityID)
+		if err != nil {
+			startupErr <- err
+			return
+		}
+		protocols = append(protocols, receiver.ProtocolForCID(CID))
+	}
+	s.mes.init(protocols...)
```

Comment:
We briefly talked about adding the protocol list to the messenger struct and let the init iterate over them? I am ok leaving it as is. The only question for me is if we see a use case on having them as part of the struct.

Reply:
Thats part of the messenger isn't it? so there won't be a change here even if we implement your suggested change.

Reply:
If we have the protocols within the messenger then we do not need to pass protocol list to the `init` method. Just `s.mes.init()`.